src/commands/mobile_app/upload.rs

```diff
@@ -77,3 +80,45 @@ fn validate_is_mobile_app(path: &Path) -> Result<()> {
         path.display()
     ))
 }
+
+/// Normalizes a mobile app file into a zip file to ensure consistent artifact parsing logic on the backend.
+fn normalize_mobile_app(path: &Path) -> Result<TempFile> {
+    let temp_file = TempFile::create()?;
+    let mut zip = ZipWriter::new(temp_file.open()?);
+
+    if path.is_file() {
+        // For APK and AAB files, we'll copy them directly into the zip
+        let mut file = File::open(path)?;
+        let file_name = path
+            .file_name()
+            .ok_or_else(|| anyhow!("Invalid file name"))?
+            .to_string_lossy();
+
+        zip.start_file(file_name, SimpleFileOptions::default())?;
+        std::io::copy(&mut file, &mut zip)?;
+    } else if path.is_dir() {
+        // For XCArchive directories, we'll zip the entire directory
+        for entry in walkdir::WalkDir::new(path)
+            .follow_links(true)
+            .into_iter()
+            .filter_map(Result::ok)
+        {
+            let entry_path = entry.path();
+            if entry_path.is_file() {
+                let relative_path = entry_path
+                    .strip_prefix(path)
+                    .map_err(|_| anyhow!("Failed to get relative path"))?;
+
+                let mut file = File::open(entry_path)?;
+                zip.start_file(
+                    relative_path.to_string_lossy(),
+                    SimpleFileOptions::default(),
+                )?;
+                std::io::copy(&mut file, &mut zip)?;
+            }
+        }
+    }
+
+    zip.finish()?;
+    Ok(temp_file)
```

Comment:
I'm new to Rust error handling but does this function handle errors properly? E.g. if opening the file or building the zip fails in some way, I thought this function would return an error result type.

Reply:
It does, the `?` at the end of `finish` is rust's error handling propogation: https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator

Reply:
Wow that is extremely subtle lol